tools/mo/openvino/tools/mo/utils/cli_parser.py

```diff
@@ -288,10 +288,15 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
     common_group.add_argument('--input',
                               help='Quoted list of comma-separated input nodes names with shapes, data types, '
                                    'and values for freezing. The shape and value are specified as space-separated '
-                                   'lists. The data type of input node is specified in braces and can have one of the '
-                                   'values: f64 (float64), f32 (float32), f16 (float16), i64 (int64), i32 (int32), u8 '
-                                   '(uint8), boolean. For example, use the following format to set input port 0 of the '
-                                   'node `node_name1` with the shape [3 4] as an input node and freeze output port 1 '
+                                   'lists. The data type of input node is specified in braces and '
+                                   'can have one of the values: f64 (float64), f32 (float32), f16 (float16), '
+                                   'i64 (int64), i32 (int32), u8 (uint8), boolean. Data type is optional and '
+                                   'in case if it is not specified then it will be derived from the original model. '
+                                   'Example, to set `input_1` with shape [1 100], `sequence_len` scalar input '
+                                   'with value `150`, and boolean input `is_training` with `False` value use the ' 
+                                   'following format: "input_1[1 10],sequence_len->150,is_training->False"'
+                                   'Another example, use the following format to set input port 0 of the node '
```

Comment:
Without explicit type specification does the current implementation creates `sequence_len` of float type instead of `int`? If yes, this is a bug in MO because `sequence_len` must be of integer type for sure. The same question for is_training.

Reply:
>Without explicit type specification does the current implementation creates sequence_len of float type instead of int? If yes, this is a bug in MO because sequence_len must be of integer type for sure.

It will not create `sequence_len` of float type. It will store as string and then will convert to the `dtype` of the original node:
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/front/freeze_placeholder_value.py#L48-L57

>The same question for is_training. I propose to automatically determine its type.
If type is not specified we take it from the original graph. The code above contains all answers.

>The previous behaviour I expect was to specify explicitly its type. Now we can make it optional but we need to do it in intelligent way.
No. Previous behavior was the same: type is optional. I didn't change behavior I only mentioned explicitly this in help.